runtime/queries/metricsview_data_types.go

```diff
@@ -0,0 +1,171 @@
+package queries
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"strings"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type MetricsViewDataTypes struct {
+	MetricsViewName    string                               `json:"metrics_view_name,omitempty"`
+	MetricsView        *runtimev1.MetricsViewSpec           `json:"-"`
+	ResolvedMVSecurity *runtime.ResolvedMetricsViewSecurity `json:"security"`
+
+	Result []*runtimev1.MetricsViewDataType `json:"-"`
+}
+
+var _ runtime.Query = &MetricsViewDataTypes{}
+
+func (q *MetricsViewDataTypes) Key() string {
+	return fmt.Sprintf("TableColumns:%s", q.MetricsViewName)
+}
+
+func (q *MetricsViewDataTypes) Deps() []*runtimev1.ResourceName {
+	return []*runtimev1.ResourceName{
+		{Kind: runtime.ResourceKindSource, Name: q.MetricsView.Table},
+		{Kind: runtime.ResourceKindModel, Name: q.MetricsView.Table},
+		{Kind: runtime.ResourceKindMetricsView, Name: q.MetricsViewName},
+	}
+}
+
+func (q *MetricsViewDataTypes) MarshalResult() *runtime.QueryResult {
+	var size int64
+	if len(q.Result) > 0 {
+		// approx
+		size = sizeProtoMessage(q.Result[0]) * int64(len(q.Result))
+	}
+	return &runtime.QueryResult{
+		Value: q.Result,
+		Bytes: size,
+	}
+}
+
+func (q *MetricsViewDataTypes) UnmarshalResult(v any) error {
+	res, ok := v.([]*runtimev1.MetricsViewDataType)
+	if !ok {
+		return fmt.Errorf("TableColumns: mismatched unmarshal input")
+	}
+	q.Result = res
+	return nil
+}
+
+func (q *MetricsViewDataTypes) Resolve(ctx context.Context, rt *runtime.Runtime, instanceID string, priority int) error {
+	olap, release, err := rt.OLAP(ctx, instanceID)
+	if err != nil {
+		return err
+	}
+	defer release()
+
+	if olap.Dialect() != drivers.DialectDuckDB {
+		return fmt.Errorf("not available for dialect '%s'", olap.Dialect())
+	}
+
+	return olap.WithConnection(ctx, priority, false, false, func(ctx context.Context, ensuredCtx context.Context, _ *sql.Conn) error {
+		// views return duplicate column names, so we need to create a temporary table
+		temporaryTableName := tempName("profile_columns_")
+
+		err = olap.Exec(ctx, &drivers.Statement{
+			Query:            fmt.Sprintf(`CREATE TEMPORARY TABLE "%s" AS (%s)`, temporaryTableName, q.buildMetricsViewDataTypesSQL()),
+			Priority:         priority,
+			ExecutionTimeout: defaultExecutionTimeout,
+		})
```

Comment:
As addressed previously, would suggest two refactors:

1. Return a `rill.runtime.v1.StructType` instead of raw type names, so we have predictable types for any OLAP backend (so the frontend doesn't get tied too closely to DuckDB type names specifically). It's available in the `Schema` property of the `drivers.Result` type returned from OLAP queries – it should be possible to construct a query that doesn't scan any data, but returns all dims and measures as fields just to get the types.
2. Make sure it also supports Druid – we need to support it as a first-class citizen for all metrics-related queries